File: express-backend/src/services/provider/GameSearchProvider.ts
import { ProviderSearchResult, ExternalGameSearchResult } from '../../dto/external-search';

// 游戏搜索 Provider 接口，与 Java 端 GameSearchProvider 对齐
export interface GameSearchProvider {
  id(): string;
  search(query: string, page: number): Promise<ProviderSearchResult<ExternalGameSearchResult>>;
}

File: legacy/java-backend/src/main/java/com/pixelreel/service/search/GameSearchProvider.java
package com.pixelreel.service.search;

import com.pixelreel.dto.external.ExternalGameSearchResult;
import com.pixelreel.dto.external.ProviderSearchResult;

public interface GameSearchProvider {
  String id();

  ProviderSearchResult<ExternalGameSearchResult> search(String query, int page);
}

